### DISCUSSION TITLE
Psycopg2 - Params provided from a compile of `in_()` can no longer be used in cursor.mogrify()

### DISCUSSION THREAD

**Discussion by rad-pat** [sqlalchemy/sqlalchemy#6230] [Usage Questions] [answered]
**Describe the bug**
In 1.4.6 - with compiled SQL & params obtained using `statement.compile()` of a statement containing `in_()`, it was possible in 1.3.* to use these directly in `cursor.mogrify()` for psycopg2 to obtain a complete SQL statement. With 1.4.*, the *in* parameters are referred to as `POSTCOMPILE_column_id`, but the param provided is simply  `column_id` and therefore does not substitute in the mogrify.

With 1.3.*, each member of the IN statement was an individual parameter.

Possibly related to https://github.com/sqlalchemy/sqlalchemy/issues/6114

**Expected behavior**
For the compile statement to return a dict of params that match the params in the compiled SQL such that this may be used with mogrify.

**To Reproduce**
Please try to provide a [Minimal, Complete, and Verifiable](http://stackoverflow.com/help/mcve) example.
See also [Reporting Bugs](https://www.sqlalchemy.org/participate.html#bugs) on the website, and some [example issues](https://github.com/sqlalchemy/sqlalchemy/issues?q=label%3A%22great+mcve%22).

**Please do not use Flask-SQLAlchemy or any other third-party extensions or dependencies in test cases**.  The test case must illustrate the problem **without** using any third party SQLAlchemy extensions.  Otherwise, please report the bug to those projects first.

```py
import sqlalchemy
from sqlalchemy import Table, Column, String, Integer, and_

metadata = sqlalchemy.MetaData()

user_table = Table(
    'user', metadata,
    Column('id', Integer, primary_key=True),
    Column('name', String(50)),
)

engine = sqlalchemy.create_engine('postgresql://...')

sel = sqlalchemy.select(
    user_table.columns
).where(
    and_(
        user_table.columns['id'] == 1,
        user_table.columns['name'].in_(['one', 'two', 'three'])
    )
)

compiled_query = sel.compile(dialect=engine.dialect)
query = str(compiled_query)
params = compiled_query.params

conn = engine.raw_connection()
cur = conn.cursor()
full_query = cur.mogrify(query, params).decode('utf-8')

print(query)
print(params)
print(full_query)

```

**Output**

```
SELECT "user".id, "user".name 
FROM "user" 
WHERE "user".id = %(id_1)s AND "user".name IN ([POSTCOMPILE_name_1])
{'id_1': 1, 'name_1': ['one', 'two', 'three']}
SELECT "user".id, "user".name 
FROM "user" 
WHERE "user".id = 1 AND "user".name IN ([POSTCOMPILE_name_1])
# ^ IN statement parameter not substituted
```

**Versions.**
 - OS: Ubuntu 18.04
 - Python: 3.9.2
 - SQLAlchemy: 1.4.6
 - Database: Postgres 10
 - DBAPI: psycopg2

**Have a nice day!**

**Comment by zzzeek** [comment, marked as answer]
HI  -

I'll add this to [the faq entry](https://docs.sqlalchemy.org/en/14/faq/sqlexpressions.html#rendering-bound-parameters-inline) momentarily:


    compiled_query = sel.compile(dialect=engine.dialect, compile_kwargs={"render_postcompile": True})

**Reply by rad-pat** [reply]
Thanks for the quick response
